doc/statemachines.tex

```diff
@@ -0,0 +1,310 @@
+
+\DeclareMathOperator{\dom}{dom}
+\DeclareMathOperator{\range}{range}
+\DeclareMathOperator*{\argmin}{arg\,min} % thin space, limits underneath in displays
+\DeclareMathOperator*{\minimum}{min}
+\DeclareMathOperator*{\maximum}{max}
+
+% Number within sections, and don't have separate counters for separate environments
+\theoremstyle{definition}{
+  \newtheorem{lemma}{Lemma}[section] % Number within sections
+  \newtheorem{definition}[lemma]{Definition}
+}
+\theoremstyle{theorem}{
+  \newtheorem{invariant}[lemma]{Invariant}
+  \newtheorem{proofobligation}[lemma]{Proof Obligation}
+}
+
+\Crefname{invariant}{Invariant}{Invariants}
+
+\numberwithin{equation}{lemma}
+
+%\floatstyle{boxed}
+%\restylefloat{figure}
+
+\lstset{basicstyle=\ttfamily\small}
+
+\raggedbottom
+
+\begin{document}
+
+\title{Specification of the Chain Synchronization Protocol\\
+       {\small (Version 0.1)} \\
+       {\large \sc An IOHK technical report}}
+\date{\today}
+
+\maketitle
+
+\begin{abstract}
+  This document describes the Chain Synchronization Protocol.
+  At this stage the focus of the document is not the accurate description of the protocol
+  but instead this document can be used to discuss the structure and layout
+  of the specifications of the IPC Protocols used in Cardano starting with the Shelley release.
+  (inside the Haskell-team and for the communication between the Rust and the Haskell team).
+  This is very much work-in-progress. 
+  Finally the description of the Chain Synchronization Protocol other IPC Protocols will be
+  become part of a comprehensive specification of the network layer.
+\end{abstract}
+
+\tableofcontents
+
+\section*{Version history}
+
+\begin{description}
+\item[Version 0.1, Dez 06, 2018]  First draft of a  document
+                                  that could be used as a basis for further discussions.
+\end{description}
+
+\section{The Cardano IPC Protocols}
+
+\subsection{Questions} \marginpar{\njd{Todo: replace the questions section with the right answers.}}
+\subsubsection{Semantic ?}
+\begin{description}
+\item[Color scheme] Todo: Use a color scheme to mark generic remarks, Haskell specific remarks and Rust
+  specific remarks.
+  Use a color scheme for comments/remarks.
+  
+\item[Do we need to define a formal/semi-formal semantics of the network layer ?]
+  Ledger specification and other parts of cardano specification, are formal  and rigorous !
+  Network protocol is equally vital for Ouroboros.
+  Which formalism are available for concurrency ?
+  What is best approachfor  efficient coordination with the Rust team ?
+  
+\item[What is the interpretation of the state machine that is defined in the typed transition framework ?]
+  The state machine is an abstract high-level description of the network protocol.
+
+\item[What questions, properties can be reasoned about in this framework ? (the typed transition framework ?)]
+  Client and server agree on the current state (in particular on the Done-state) ?
```

Comment:
Yes, client and the server when they are implemented in the typed-transition framework are guaranteed to agree on state.  But this is a Haskell implementation detail.